src/components/Characters.js

```diff
@@ -0,0 +1,66 @@
+import { useEffect, useState } from "react";
+import { Link } from "react-router-dom";
+
+export default function Characters() {
+  const [characters, setCharacters] = useState([]);
+  const [list, setList] = useState(12);
+
+
+  useEffect(() => {
+    const url = `https://akabab.github.io/starwars-api/api/all.json`;
+
+    fetch(url)
+      .then((res) => res.json())
+      .then((dataFromApi) => {
+        setCharacters((prevCharacters) => {
+         return [...prevCharacters, ...dataFromApi];
+
+      });
+
+
+    console.log(dataFromApi)
+      });
+
+  }, []);
+  function renderCharacters() {
+  return characters.filter((person) => person.id <= list).map((person) => {
+  const {id, name, image} = person;
+
+ return (
+
+
+        <section key={name} class="characterCard">
```

Comment:
not too sure, but it looks like you receive an id from your API (line 27) so i'd set the key to {id} to make it more specific.